File: Lexor/Lexor.Model/Responses/NotificationResponse.cs
using Lexor.Model.Enums;

namespace Lexor.Model.Responses
{
    public class NotificationResponse
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Body { get; set; } = string.Empty;
        public NotificationType NotificationType { get; set; }
        public bool IsRead { get; set; }
        public DateTime CreatedAt { get; set; }
        public RelatedEntityType? RelatedEntityType { get; set; }
        public int? RelatedEntityId { get; set; }

    }
}

File: Lexor/UI/lexor_mobile/lib/models/notification_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'notification_response.g.dart';

@JsonSerializable(createToJson: false)
class NotificationResponse {
  final int id;
  final String title;
  final String body;
  bool isRead; // mutable so we can flip it locally after marking as read
  final DateTime createdAt;

  NotificationResponse({
    required this.id,
    required this.title,
    required this.body,
    required this.isRead,
    required this.createdAt,
  });

  factory NotificationResponse.fromJson(Map<String, dynamic> json) =>
      _$NotificationResponseFromJson(json);
}
